### PR TITLE
BaSP-MR-09-01: Get & Get by Id Super admin

### PR DESCRIPTION
**Pull request by Flor-Farruggia**
Request created for Super Admin:
Get: to be able to see all the super admin list.
Get by id: to be able to obtain an specific super admin.

### CODE REVIEW

File: src/controllers/super-admin.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  } = req.body;
+  superAdmin.create({
+    email,
+    password,
+  })
+    .then((result) => res.status(201).json({
+      message: 'Super Admin created',
+      result,
+    }))
+    .catch((error) => res.status(400).json({
```
Comment (NicoCagnina) [inline src/controllers/super-admin.js:15]: this should be 500

File: src/controllers/super-admin.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  superAdmin.find()
+    .then((data) => {
+      if (data) {
+        res.status(200).json({
+          message: 'This are all the super-admin',
+          data,
+        });
+      }
+    })
+    .catch((error) => res.status(404).json({
```
Comment (NicoCagnina) [inline src/controllers/super-admin.js:31]: this should be 500

**Review by NicoCagnina — CHANGES_REQUESTED** [review]

File: src/validations/super-admin.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
+const Joi = require('joi');
+
+const validateUpdate = (req, res, next) => {
+  const superAdminValidation = Joi.object({
+    email: Joi.string()
+      .email()
+      .required(),
```
Comment (JuanMornik) [inline src/validations/super-admin.js:7]: Inputs aren´t required on update validations

File: src/validations/super-admin.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    password: Joi.string()
+      .min(8)
+      .pattern(/^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)[a-zA-Z\d]{8,}$/)
+      .required()
```
Comment (JuanMornik) [inline src/validations/super-admin.js:11]: delete this require too

**Review by JuanMornik — CHANGES_REQUESTED** [review]

**Review by JuanMornik — APPROVED** [review]

**Review by NicoCagnina — APPROVED** [review]

**Pull request merged**